File: ADT/trees/segment_tree/3477.cpp
// Tc = O(n^2)
// Sc = O(1)

class Solution {
public:
    int numOfUnplacedFruits(vector<int>& fruits, vector<int>& baskets) {
        int n = fruits.size();
        int unplaced = 0;
        for (int i = 0; i < n; i++) {
            bool placed = false;
            for (int j = 0; j < n; j++) {
                if (fruits[i] <= baskets[j]) {
                    placed = true;
                    baskets[j] = -1;
                    break;
                }
            }
            if (!placed) {
                unplaced++;
            }
        }
        return unplaced;
    }
};

File: ADT/trees/segment_tree/3477.py
class Solution:
    def numOfUnplacedFruits(self, fruits: List[int], baskets: List[int]) -> int:
        n = len(fruits)
        unplaced = 0
        for i in range(n):
            placed = False
            for j in range(n):
                if fruits[i] <= baskets[j]:
                    placed = True
                    baskets[j] = -1
                    break
            if not placed:
                unplaced += 1
        return unplaced
